app/cdap/components/GridContainer/index.tsx

```diff
@@ -14,64 +14,119 @@
+      }, 5000);
+    }
+  }, [snackbarState.open]);
 
-    // check if both function and column are selected
-    if (
-      (valueToUpdate == 'function' && transformationPayload.column) ||
-      (valueToUpdate == 'column' && transformationPayload.function)
-    ) {
-      // if yes, then do applyDirective
-      const directive = getDirective(newValue.option, transformationPayload.column);
-      // addDirectives()
+  /*
+   * In this useEffect we are handling apply transformation based on conditions
+   * whether to apply transformation directly without add transformation panel open
+   * or should pass data to add transformation step panel
+   */
+  useEffect(() => {
+    // checks if both function and column are selected
+    if (selectedColumn && selectedFunction) {
+      // If both function and column are selected, then apply transformation
+      const directive = getDirective(selectedFunction.option, selectedColumn);
       execute([directive]).subscribe(
-        () => {},
-        (error) => {
-          DataPrepStore.dispatch({
-            type: DataPrepActions.setError,
-            payload: {
-              message: error.message || error.response.message,
+        () => {
+          setSelectedColumn(false);
+          setSelectedFunction({
+            option: {
+              getUsage({}) {
+                return '';
+              },
+              label: '',
+              supportedDataType: [''],
+              value: '',
             },
+            supportedType: [''],
+          });
+          setSnackbar({
+            open: true,
+            isSuccess: true,
+            message: 'Transformation Successfully Applied',
+          });
+        },
+        (error) => {
+          setSnackbar({
+            open: true,
+            isSuccess: false,
+            message: error.message || error.response.message,
           });
         }
       );
+    } else {
+      // If selectedColumn is not selected then open add transformation panel and apply transformation
     }
-    // if no then return, do nothing
-  };
+  }, [selectedFunction, selectedColumn]);
 
-  const getDirective = (functionName: string, selectedColumnName: string | boolean) => {
-    if (DATATYPE_OPTIONS.some((eachOption) => eachOption.value === functionName)) {
-      return `set-type :${selectedColumnName} ${functionName}`;
+  const handleSelectedFunctionColumnState = (valueToUpdate: string, newValue) => {
+    if (valueToUpdate === 'function') {
+      setSelectedFunction(newValue);
+    } else if (valueToUpdate === 'column') {
+      setSelectedFunction(null);
+      setSelectedColumn(newValue);
     }
   };
 
   return (
-      <Provider store={DataPrepStore}>
-        <GridTable handleTransformationUpload={handleTransformationUpload} storeData={storeData} />
-      </Provider>
+    <Provider store={DataPrepStore}>
+      <GridTable
+        handleSelectedFunctionColumnState={handleSelectedFunctionColumnState}
+        storeData={storeData}
```

Comment:
check if we need store data or not, we may be able to use useSelector instead
does it make any performance difference?

Reply:
Instead of connect method used useSelector hook